pkg/machine/hyperv/volumes.go

```diff
@@ -40,15 +42,27 @@ func removeShares(mc *vmconfigs.MachineConfig) error {
 
 func startShares(mc *vmconfigs.MachineConfig) error {
 	for _, mount := range mc.Mounts {
-		args := []string{"-q", "--", "sudo", "podman"}
+		args := []string{"-q", "--"}
+
+		cleanTarget := path.Clean(mount.Target)
+		requiresChattr := !strings.HasPrefix(cleanTarget, "/home") && !strings.HasPrefix(cleanTarget, "/mnt")
+		if requiresChattr {
+			args = append(args, "sudo", "chattr", "-i", "/", "; ")
+		}
+		args = append(args, "sudo", "mkdir", "-p", cleanTarget, "; ")
+		if requiresChattr {
+			args = append(args, "sudo", "chattr", "+i", "/", "; ")
+		}
+
+		args = append(args, "sudo", "podman")
 		if logrus.IsLevelEnabled(logrus.DebugLevel) {
 			args = append(args, "--log-level=debug")
 		}
 		// just being protective here; in a perfect world, this cannot happen
 		if mount.VSockNumber == nil {
 			return errors.New("cannot start 9p shares with undefined vsock number")
 		}
-		args = append(args, "machine", "client9p", fmt.Sprintf("%d", mount.VSockNumber), mount.Target)
+		args = append(args, "machine", "client9p", fmt.Sprintf("%d", *mount.VSockNumber), mount.Target)
```

Comment:
Another reason why using fmt.Sprintf() is bad, if you would use `strconv.Itoa`/ `strconv.FormatInt` this would not have been possible.